server/text_generation_server/models/causal_lm.py

```diff
@@ -632,7 +632,7 @@ def __init__(
             model = model.eval().to(device)
             #wrap in hpu_graph only if self.enable_hpu_graph is set
             if self.enable_hpu_graph:
-                model = wrap_in_hpu_graph(model)
+                model = wrap_in_hpu_graph(model, disable_tensor_cache=True)
```

Comment:
did you try the LIMITED_HPU_GRAPH=True?

Reply:
Yes, we do it f.e. when running llama-70b - this is separate mechanism. Introduced one allows to reduce memory additionally.

Reply:
can you have option to change it based on cmd line if customer is running smaller model?